analyzers/tests/SonarAnalyzer.Test/SymbolicExecution/Roslyn/RoslynSymbolicExecutionTest.Loops.cs

```diff
@@ -25,6 +25,244 @@ namespace SonarAnalyzer.Test.SymbolicExecution.Roslyn;
+            """;
+        var validator = SETestContext.CreateCS(code).Validator;
+        validator.PostProcessed.GroupBy(x => x.Block.Ordinal).Should().HaveCount(7);
+        validator.PostProcessed.Where(x => x.Block.Ordinal is 2 or 3 or 4 or 5 or 6).Should().AllSatisfy(x => x.IsInLoop.Should().BeTrue());
+        validator.PostProcessed.Where(x => x.Block.Ordinal is 1 or 7).Should().AllSatisfy(x => x.IsInLoop.Should().BeFalse());
+    }
+
+    [TestMethod]
+    public void Loops_IsInLoop_TryCatch()
+    {
+        var code = """
+            Console.WriteLine("Before loop");               // Block 1
+            while (Condition)                               // Block 2
+            {
+                Console.WriteLine("Inside loop");           // Block 3
+                try
+                {
+                    Console.WriteLine("Inside try");        // Block 4
+                }
+                catch (Exception)
+                {
+                    Console.WriteLine("Inside catch");      // Block 5
+                }
+                finally
+                {
+                    Console.WriteLine("Inside finally");    // Block 6
+                }
+            }
+            Console.WriteLine("After loop");                // Block 7
+            """;
+        var validator = SETestContext.CreateCS(code).Validator;
+        validator.PostProcessed.GroupBy(x => x.Block.Ordinal).Should().HaveCount(7);
+        validator.PostProcessed.Where(x => x.Block.Ordinal is 2 or 3 or 4).Should().AllSatisfy(x => x.IsInLoop.Should().BeTrue());      // should be 2-6
+        validator.PostProcessed.Where(x => x.Block.Ordinal is 1 or 5 or 6 or 7).Should().AllSatisfy(x => x.IsInLoop.Should().BeFalse());
+    }
+
+    [TestMethod]
+    public void Loops_IsInLoop_Complex()
+    {
+        var code = """
+            if (Condition)                          // Block 1
+            {
+                goto AlternativeEntry;
+            }
+            Loop:
+            if (Condition)                          // Block 2
+            {
+                goto AfterLoop;
+            }
+            AlternativeEntry:
+            if (Condition)                          // Block 3
+            {
+                goto Skip;
+            }
+            Console.WriteLine("Skippable");         // Block 4
+            Skip:
+            if (Condition)                          // Block 5
+            {
+                goto Loop;
+            }
+            AfterLoop:
+            Console.WriteLine("After loop");        // Block 6
+            """;
+        var validator = SETestContext.CreateCS(code).Validator;
+        validator.PostProcessed.GroupBy(x => x.Block.Ordinal).Should().HaveCount(6);
+        validator.PostProcessed.Where(x => x.Block.Ordinal is 2 or 3 or 4 or 5).Should().AllSatisfy(x => x.IsInLoop.Should().BeTrue());
+        validator.PostProcessed.Where(x => x.Block.Ordinal is 1 or 6).Should().AllSatisfy(x => x.IsInLoop.Should().BeFalse());
```

Comment:
I would add a couple more cases, just to be safe about the shape of the CFG:

- while/for with two sequential ifs
- while/for with two _nested_ ifs
- if-while/for-if